tests/thread/host/spinlock_test_host.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <signal.h>
+#include <stdio.h>
+#include <cassert>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <thread>
+#include "thread_u.h"
+
+#define ERRNO_TEST_TIMES 10
+#define LARGE_ENOUGH_MEMORY_SIZE 1024 * 1024 * 1024
+#define NUM_SPINLOCK_TEST_THREADS 2
+
+int g_flag_withlock = 0;
+static uint32_t g_lock = 0;
+static int g_error_num = 0;
+
+void* errno_thread1(void* para)
+{
+    int i = 0;
+    int t_errno = 0;
+    void* buffer = NULL;
+    oe_enclave_t* enclave = (oe_enclave_t*)para;
+
+    for (i = 0; i < ERRNO_TEST_TIMES; i++)
+    {
+        if (g_flag_withlock)
+        {
+            enc_spin_lock(enclave, &g_lock);
+        }
+
+        enc_malloc(enclave, &buffer, LARGE_ENOUGH_MEMORY_SIZE, &t_errno);
+
+        if (ENOMEM != t_errno)
+        {
+            printf(
+                "[errno_thread1]Error check errno, expected %d, got %d.\n",
+                ENOMEM,
+                t_errno);
+            g_error_num |= 0x8;
+            if (g_flag_withlock)
+            {
+                enc_spin_unlock(enclave, &g_lock);
```

Comment:
After this `enc_spin_unlock` executes, the unlock in line 57 would execute too leading to double unlock.